src/dc_sqlite3.rs

```diff
@@ -725,58 +725,10 @@ pub unsafe fn dc_sqlite3_open(
-                                        b"dc_sqlite3_open\x00",
-                                    ))
-                                    .as_ptr(),
-                                    b"../src/dc_sqlite3.c\x00" as *const u8 as *const libc::c_char,
-                                    559,
-                                    b"DC_MOVE_STATE_UNDEFINED == 0\x00" as *const u8
-                                        as *const libc::c_char,
-                                );
-                            } else {
-                            };
-                            if 0 != !(DC_MOVE_STATE_PENDING as libc::c_int == 1) as usize {
-                                __assert_rtn(
-                                    (*::std::mem::transmute::<&[u8; 16], &[libc::c_char; 16]>(
-                                        b"dc_sqlite3_open\x00",
-                                    ))
-                                    .as_ptr(),
-                                    b"../src/dc_sqlite3.c\x00" as *const u8 as *const libc::c_char,
-                                    560,
-                                    b"DC_MOVE_STATE_PENDING == 1\x00" as *const u8
-                                        as *const libc::c_char,
-                                );
-                            } else {
-                            };
-                            if 0 != !(DC_MOVE_STATE_STAY as libc::c_int == 2) as usize {
-                                __assert_rtn(
-                                    (*::std::mem::transmute::<&[u8; 16], &[libc::c_char; 16]>(
-                                        b"dc_sqlite3_open\x00",
-                                    ))
-                                    .as_ptr(),
-                                    b"../src/dc_sqlite3.c\x00" as *const u8 as *const libc::c_char,
-                                    561,
-                                    b"DC_MOVE_STATE_STAY == 2\x00" as *const u8
-                                        as *const libc::c_char,
-                                );
-                            } else {
-                            };
-                            if 0 != !(DC_MOVE_STATE_MOVING as libc::c_int == 3) as usize {
-                                __assert_rtn(
-                                    (*::std::mem::transmute::<&[u8; 16], &[libc::c_char; 16]>(
-                                        b"dc_sqlite3_open\x00",
-                                    ))
-                                    .as_ptr(),
-                                    b"../src/dc_sqlite3.c\x00" as *const u8 as *const libc::c_char,
-                                    562,
-                                    b"DC_MOVE_STATE_MOVING == 3\x00" as *const u8
-                                        as *const libc::c_char,
-                                );
-                            } else {
-                            };
+                            assert_eq!(DC_MOVE_STATE_UNDEFINED, 0);
```

Comment:
what do these asserts check, arent they comparing a constant to a hardcoded number?

Reply:
Aye, they assert values on defines https://github.com/deltachat/deltachat-core/blob/master/src/dc_sqlite3.c#L559-L562

Reply:
I guess we can remove them altogether.

Reply:
> what do these asserts check, arent they comparing a constant to a hardcoded number?

yip. for two reasons:
- sometimes, in the sql-strings, the number is used directly instead of the content (for $reasons in c) - the assert is a hint about what the number represents.
- more important, the asserts make sure, the numbers are not changed accidentially by some refactoring or so - the program will work and new installations as well, however, as the numbers are also used in the database, these changes will break compatibility to existing installation